src/MaaCore/Task/Roguelike/RoguelikeRecruitTaskPlugin.cpp

```diff
@@ -370,6 +382,93 @@ bool asst::RoguelikeRecruitTaskPlugin::check_char(const std::string& char_name,
     return false;
 }
 
+bool asst::RoguelikeRecruitTaskPlugin::check_support_char()
+{
+    LogTraceFunction;
+    const int MaxRefreshTimes = 3;
+
+    auto core_opt = status()->get_str(Status::RoguelikeCoreChar);
+    status()->set_str(Status::RoguelikeCoreChar, "");
+    if (core_opt && !core_opt->empty()) {
+        if (check_support_char(core_opt.value(), MaxRefreshTimes)) return true;
+    }
+    return false;
+}
+
+bool asst::RoguelikeRecruitTaskPlugin::check_support_char(const std::string& name, const int max_refresh)
+{
+    LogTraceFunction;
+
+    // 判断是否存在“选择助战”按钮，存在则点击
+    auto screen_choose = ctrler()->get_image();
+    auto analyzer_choose = RoguelikeRecruitSupportAnalyzer(screen_choose);
```

Comment:
虽说这样写是没什么问题啦，只是不太符合 C++ 一般的习惯
```suggestion
    RoguelikeRecruitSupportAnalyzer analyzer_choose(screen_choose);
```